Added feature to filter subscriptions based on teams

Kshitij-Katiyar opened the pull request:
#### Summary
Added feature to filter subscriptions based on teams

#### Screenshots
![Screenshot from 2025-04-16 14-10-02](https://github.com/user-attachments/assets/0ebfbf88-8549-41b4-86ca-0f8c1eae6ab9)
![Screenshot from 2025-04-16 14-09-51](https://github.com/user-attachments/assets/474cbf63-03a4-47b9-a063-cf70da9e6fed)
![Screenshot from 2025-04-16 14-10-48](https://github.com/user-attachments/assets/749542b1-ca02-4a14-9b96-043507febff2)

wiggin77 reviewed (changes requested):
see comment

Kshitij-Katiyar commented on plugin.json:
```diff
@@ -140,6 +140,14 @@
+                "key": "TeamIDs",
+                "display_name": "Team List",
+                "type": "text",
+                "help_text": "comma separated list of team ID's and team name to be used to filter subscriptions. Use the format (team-name)[team-id]",
```
Sure we can, will do

Kshitij-Katiyar replied:
Updated

wiggin77 reviewed (approved):
LGTM :+1:

raghavaggarwal2308 commented on plugin.json:
```diff
@@ -140,6 +140,14 @@
+                "key": "TeamIDs",
+                "display_name": "Team List",
+                "type": "text",
+                "help_text": "Comma separated list of team name and IDs to be used for filter subscriptions",
```
```suggestion
                "help_text": "Comma separated list of team name and IDs to be used for filtering subscriptions",
```

raghavaggarwal2308 commented on server/plugin.go:
```diff
@@ -205,6 +221,34 @@ func (p *Plugin) OnConfigurationChange() error {
+			teamName := strings.TrimSpace(match[1])
+			teamID := strings.TrimSpace(match[2])
+
+			if len(teamName) == 0 {
```
We should not skip if the id is wrong

Kshitij-Katiyar commented:
@AayushChaudhary0001 Please test this PR

AayushChaudhary0001 commented:
While testing this PR, few issues were found which are listed below:-

- No error is shown when providing incorrect details in the plugin configuration settings.

- Team -> Empty: No notifications are triggered for issues created with no team assigned.

- [Enhancement] : A validation check can be added to prevent a user from adding a filter that contradicts an existing filter.

AayushChaudhary0001 commented:
I found a scenario while retesting in which no notifications are triggered when plugin config setting for draft PR is disabled and a draft PR is marked for review with complete details.

![image](https://github.com/user-attachments/assets/9ae8c840-d18c-4713-b5a2-b1f0aac1b24f)


No notification:- 

![image](https://github.com/user-attachments/assets/f6b4f927-2a3f-456f-85e8-80c6cf679f8e)

Kshitij-Katiyar commented:
@AayushChaudhary0001 I have fixed the issues.
- Errors are shown in the server logs, and we can't show the error on the UI
- Fixed the empty filter.
- Adding a check for contradicting filters is too much manual work and is quite a visible issue that users can avoid.

AayushChaudhary0001 commented:
While retesting below issue is found:
- Still no error is shown when providing incorrect details in the plugin configuration settings for team list in the server logs.

Kshitij-Katiyar commented:
@AayushChaudhary0001 I have fixed the issue, please re-test

AayushChaudhary0001 commented:
The error currently appears only in a few cases and not across all scenarios. Can it be modified to consistently display an error for all instances where either a team ID or team name is not provided correctly within the defined brackets and separated by commas?

AayushChaudhary0001 reviewed (approved):
The PR is tested, and code changes are working fine on adding threaded  comment subscription post for both local jira server and cloud jira instances for local and cloud MM.

LGTM, Approved
